astropy/io/fits/tests/test_hdulist.py

```diff
@@ -913,3 +914,38 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         # This should raise an IOError because there is no end card.
         with pytest.raises(IOError):
             fits.open(filename)
+
+    # Avoid a spurious pyflakes warning about an unused import
+    # of six by adding a reason string
+    @pytest.mark.skipif(six.PY2, reason='six.PY2')
+    @pytest.mark.parametrize('ignore', [False, True])
+    @pytest.mark.parametrize('fileobj', [False, True])
```

Comment:
Response to @saimn 's comment:

Well, I realized that I really should check that the `ignore_missing_end` behavior is checked when a user has explicitly opened the file. Given the use of `pytest.mark.parametrize` I realized that I could get all versions written in a single - but ugly - test.